lib/spoom/sorbet/metrics.rb

```diff
@@ -0,0 +1,104 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Spoom
+  module Sorbet
+    class Metrics < T::Struct
+      extend T::Sig
+
+      DEFAULT_PREFIX = "ruby_typer.unknown.."
+      SIGILS = T.let(["ignore", "false", "true", "strict", "strong", "__STDLIB_INTERNAL"], T::Array[String])
+
+      const :repo, String
+      const :sha, String
+      const :status, String
+      const :branch, String
+      const :timestamp, Integer
+      const :uuid, String
+      const :metrics, T::Hash[String, T.nilable(Integer)]
+
+      sig { params(path: String, prefix: String).returns(Metrics) }
+      def self.parse_file(path, prefix = DEFAULT_PREFIX)
+        parse_string(File.read(path), prefix)
+      end
+
+      sig { params(string: String, prefix: String).returns(Metrics) }
+      def self.parse_string(string, prefix = DEFAULT_PREFIX)
+        parse_json(JSON.parse(string), prefix)
+      end
+
+      sig { params(obj: T::Hash[String, T.untyped], prefix: String).returns(Metrics) }
+      def self.parse_json(obj, prefix = DEFAULT_PREFIX)
+        Metrics.new(
+          repo: obj.fetch("repo"),
+          sha: obj.fetch("sha"),
+          status: obj.fetch("status"),
+          branch: obj.fetch("branch"),
+          timestamp: obj.fetch("timestamp").to_i,
+          uuid: obj.fetch("uuid"),
+          metrics: obj["metrics"].each_with_object({}) do |metric, all|
+            name = metric["name"]
+            name = name.sub(prefix, '')
+            all[name] = metric["value"].to_i
+          end,
+        )
+      end
+
+      sig { returns(T::Hash[String, T.nilable(Integer)]) }
+      def files_by_sigils
+        SIGILS.each_with_object({}) do |sigil, map|
+          map[sigil] = metrics["types.input.files.sigil.#{sigil}"]
+        end
+      end
+
+      sig { returns(Integer) }
+      def files_count
+        count = 0
+        files_by_sigils.each { |_, v| count += v if v }
+        count
```

Comment:
How does this compare to something like `files_by_sigils.keys.compact.count`? Are the chained methods slower?

Reply:
I believe we would translate `files_by_sigils.keys.compact.count` as:

```ruby
        count = 0
        files_by_sigils.each { |k, _| count += 1 }
        count
```

Which is not what we want here.

But we can do this:

```ruby
files_by_strictness.values.compact.sum
```